### PR TITLE
selftests/deployment: Using Ansible' roles feature

### PR DESCRIPTION
**Pull request by beraldoleal**
The current playbook layout was not following the Ansible best practices for playbooks.  This small refactoring uses roles to improve the organization of playbooks and facilitate future reuse. For now, it is just applied to pip-git.

Also fixed small warnings.

### CODE REVIEW

File: selftests/deployment/README [hunk 1]
```diff
@@ -41,4 +41,4 @@ All-in-one execution in container:
 It may be useful to run a playbook in a fresh container.  One example
 of a onliner that can achieve that:
```
Comment (clebergnu) [inline selftests/deployment/README]: Not related to this, but this should read "oneliner" instead.

File: selftests/deployment/README [hunk 2]
```diff
@@ -41,4 +41,4 @@ All-in-one execution in container:
...
 
-  podman run --rm -ti fedora:30 /bin/bash -c 'dnf -y install git ansible && ansible-pull -v -U git://github.com/avocado-framework/avocado -i selftests/deployment/inventory -c local selftests/deployment/rpm-copr.yml'
+  podman run --rm -ti fedora:30 /bin/bash -c 'dnf -y install git ansible python3-pyyaml && ansible-pull -v -U git://github.com/avocado-framework/avocado -i selftests/deployment/inventory -c local selftests/deployment/rpm-copr.yml'
```
Comment (clebergnu) [inline selftests/deployment/README]: I can not reproduce this.  I've tried it on Fedora 30, 31 and rawhide.

Comment (beraldoleal) [reply selftests/deployment/README]: And about with:

`ansible-playbook pip-git.yml`

It works?

Comment (clebergnu) [reply selftests/deployment/README]: Yes... and in fact, `python3-pyyaml` gets pulled at `ansible` install time.  It's in the package requirements:

```
# rpm -qR ansible
/usr/bin/python3
config(ansible) = 2.9.1-1.fc31
python(abi) = 3.7
python3-jinja2
python3-jmespath
python3-setuptools
python3-six
python3.7dist(cryptography)
python3.7dist(jinja2)
python3.7dist(pyyaml)
rpmlib(CompressedFileNames) <= 3.0.4-1
rpmlib(FileDigests) <= 4.6.0-1
rpmlib(PartialHardlinkSets) <= 4.0.4-1
rpmlib(PayloadFilesHavePrefix) <= 4.0-1
rpmlib(PayloadIsZstd) <= 5.4.18-1
sshpass

rpm -q --whatprovides 'python3.7dist(pyyaml)'
python3-pyyaml-5.1.2-1.fc31.x86_64
```

Comment (beraldoleal) [reply selftests/deployment/README]: @clebergnu I was looking into this PR changes and I couldn't find any change related to `rpm-copr.yml`. I'm just changing the `pip-git.yml` logic. 

Was the problem with the `rpm-copr`, or with `python3-pyyaml`?  It was introduced by this PR?

Comment (beraldoleal) [reply selftests/deployment/README]: For some reason, when on fedora 30 python3-pyyaml was not installed after ansible installation. Strange.

Comment (beraldoleal) [reply selftests/deployment/README]: Forget about it. My mistake. It was on fedora 29. I will remove the pyyaml package on v2.

**Comment by clebergnu** [comment]
Hi @beraldoleal, I failed to say that before, the general direction here looks good.  Please proceed with a non-draft PR.

**Pull request closed without merging**